### ISSUE TITLE
A few problems building/running the benchmarks.

### ISSUE THREAD

**Issue by tkvangorder** [dsyer/spring-boot-startup-bench#4]
First off, this is a really cool project. 

I know this is an "experimental" project, so I am not expecting any "fixes" for any of these issues and more just looking for guidance/validation to how I worked around those issues. 

- The maven build was failing in a couple of spots, due to unresolved dependencies with the plugins. I was able to fix those problems by adding `	<pluginRepositories>` for spring-snapshots, spring-milestones, and spring-releases. I had to do this in several of the sub-modules (alt, static, etc) I suspect this is just that those repositories/dependencies are part of your local setup. 

Once I put the following in most of the pom.xml files, things built cleanly: 

```
	<repositories>
		<repository>
			<id>spring-snapshots</id>
			<name>Spring Snapshots</name>
			<url>https://repo.spring.io/libs-snapshot-local</url>
			<snapshots>
				<enabled>true</enabled>
			</snapshots>
		</repository>
		<repository>
			<id>spring-milestones</id>
			<name>Spring Milestones</name>
			<url>https://repo.spring.io/libs-milestone-local</url>
			<snapshots>
				<enabled>false</enabled>
			</snapshots>
		</repository>
		<repository>
			<id>spring-releases</id>
			<name>Spring Releases</name>
			<url>https://repo.spring.io/release</url>
			<snapshots>
				<enabled>false</enabled>
			</snapshots>
		</repository>
	</repositories>
	<pluginRepositories>
		<pluginRepository>
			<id>spring-snapshots</id>
			<name>Spring Snapshots</name>
			<url>https://repo.spring.io/libs-snapshot-local</url>
			<snapshots>
				<enabled>true</enabled>
			</snapshots>
		</pluginRepository>
		<pluginRepository>
			<id>spring-milestones</id>
			<name>Spring Milestones</name>
			<url>https://repo.spring.io/libs-milestone-local</url>
			<snapshots>
				<enabled>false</enabled>
			</snapshots>
		</pluginRepository>
		<pluginRepository>
			<id>spring-releases</id>
			<name>Spring Releases</name>
			<url>https://repo.spring.io/release</url>
			<snapshots>
				<enabled>false</enabled>
			</snapshots>
		</pluginRepository>
	</pluginRepositories>
```

- After a successful build, I tried running the root benchmarks and some of the individual tests failed due to missing resources.  I realized there were a couple of project (like configserver) where it only generated one version of application (14x) and it was missing 13x & 15x. I figured out I needed to just run with the different profiles (which is what it looks like the "alt" sub-module is doing). I didnt dig much deeper to figure out what was going on, I am running the benchmarks again and capturing the entire log. I will investigate those results on a different weekend. 8) 

- In the static sub-module: I was able to run "main", "strip", and "snap" benchmarks, but the "old" benchmarks are "frozen" after the dependency resolution logging. No CPU usage after this freeze...it's a strange one and the last thing I see in the logs:

```

2017-10-01 14:53:42.327 [com.example.bench.OldBenchmark.old-jmh-worker-1] INFO  o.s.b.loader.thin.DependencyResolver - Resolved: org.slf4j:jcl-over-slf4j:1.7.25=/Users/tyler.vangorder/.m2/repository/org/slf4j/jcl-over-slf4j/1.7.25/jcl-over-slf4j-1.7.25.jar
2017-10-01 14:53:42.327 [com.example.bench.OldBenchmark.old-jmh-worker-1] INFO  o.s.b.loader.thin.DependencyResolver - Resolved: org.springframework:spring-core:4.3.10.RELEASE=/Users/tyler.vangorder/.m2/repository/org/springframework/spring-core/4.3.10.RELEASE/spring-core-4.3.10.RELEASE.jar
```

**Comment by dsyer** [comment]
Thanks for the update. 

* I fixed the repository declarations (you are right, I have them in my local settings) in 468b94f.

* The 13x and 15x artifacts are supposed to be generated by the "alt" module, which is included in the main reactor build, so "mvn clean install" ought to be enough to make it work. I have had some bugs where artifacts got overwritten. Maybe it's still doing that (fixed in 2774465).

* The "old" benchmark experience is more worrying. I'll also look into that.

**Comment by tkvangorder** [comment]
Thanks for the quick turn around, I have no idea how you balance your workload with all the feedback you are constantly dealing with!

I suspect, I was running into the same threading issue with the old benchmarks, as I encountered with the overall benchmark test. I had to end up killing the forked process. I did capture a thread dump before killing on of the processes. 

The jmh-worker-1, its a daemon thread..but looks like the main thread is waiting on it. And it looks to be blocked on itself, maybe some interplay to launching a process...pretty strange. 
```
   java.lang.Thread.State: WAITING (on object monitor)
	at java.lang.Object.wait(Native Method)
	- waiting on <0x000000076b432238> (a java.lang.UNIXProcess)
	at java.lang.Object.wait(Object.java:502)
	at java.lang.UNIXProcess.waitFor(UNIXProcess.java:395)
	- locked <0x000000076b432238> (a java.lang.UNIXProcess)
	at com.example.ProcessLauncherState.capture(ProcessLauncherState.java:74)
```

And the whole thread dump for completeness.

```java
Full thread dump Java HotSpot(TM) 64-Bit Server VM (25.131-b11 mixed mode):

"Attach Listener" #14 daemon prio=9 os_prio=31 tid=0x00007fc83d02a000 nid=0x130f waiting on condition [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

"process reaper" #13 daemon prio=10 os_prio=31 tid=0x00007fc83c8a5800 nid=0x5c03 runnable [0x0000700001600000]
   java.lang.Thread.State: RUNNABLE
	at java.lang.UNIXProcess.waitForProcessExit(Native Method)
	at java.lang.UNIXProcess.lambda$initStreams$3(UNIXProcess.java:289)
	at java.lang.UNIXProcess$$Lambda$9/692924682.run(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:748)

"com.example.SpringBootThinBenchmark.basic14xPrecomputeThin-jmh-worker-1" #12 daemon prio=5 os_prio=31 tid=0x00007fc83c8a2800 nid=0x5a0f in Object.wait() [0x00007000015d4000]
   java.lang.Thread.State: WAITING (on object monitor)
	at java.lang.Object.wait(Native Method)
	- waiting on <0x000000076b432238> (a java.lang.UNIXProcess)
	at java.lang.Object.wait(Object.java:502)
	at java.lang.UNIXProcess.waitFor(UNIXProcess.java:395)
	- locked <0x000000076b432238> (a java.lang.UNIXProcess)
	at com.example.ProcessLauncherState.capture(ProcessLauncherState.java:74)
	at com.example.SpringBootThinBenchmark$Basic14xPrecomputeThinState.compute(SpringBootThinBenchmark.java:103)
	at com.example.generated.SpringBootThinBenchmark_basic14xPrecomputeThin_jmhTest._jmh_tryInit_f_basic14xprecomputethinstate1_G(SpringBootThinBenchmark_basic14xPrecomputeThin_jmhTest.java:587)
	- locked <0x000000076b0fe328> (a java.lang.Class for com.example.generated.SpringBootThinBenchmark_basic14xPrecomputeThin_jmhTest)
	at com.example.generated.SpringBootThinBenchmark_basic14xPrecomputeThin_jmhTest.basic14xPrecomputeThin_AverageTime(SpringBootThinBenchmark_basic14xPrecomputeThin_jmhTest.java:199)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.openjdk.jmh.runner.BenchmarkHandler$BenchmarkTask.call(BenchmarkHandler.java:464)
	at org.openjdk.jmh.runner.BenchmarkHandler$BenchmarkTask.call(BenchmarkHandler.java:448)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:748)

"Service Thread" #9 daemon prio=9 os_prio=31 tid=0x00007fc83c01a000 nid=0x5203 runnable [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

"C1 CompilerThread3" #8 daemon prio=9 os_prio=31 tid=0x00007fc83b824000 nid=0x5003 waiting on condition [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

"C2 CompilerThread2" #7 daemon prio=9 os_prio=31 tid=0x00007fc83c81d000 nid=0x4e03 waiting on condition [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

"C2 CompilerThread1" #6 daemon prio=9 os_prio=31 tid=0x00007fc83c029000 nid=0x4c03 waiting on condition [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

"C2 CompilerThread0" #5 daemon prio=9 os_prio=31 tid=0x00007fc83d02d800 nid=0x4a03 waiting on condition [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

"Signal Dispatcher" #4 daemon prio=9 os_prio=31 tid=0x00007fc83d029800 nid=0x3b0f runnable [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

"Finalizer" #3 daemon prio=8 os_prio=31 tid=0x00007fc83b823000 nid=0x3803 in Object.wait() [0x0000700000d3a000]
   java.lang.Thread.State: WAITING (on object monitor)
	at java.lang.Object.wait(Native Method)
	- waiting on <0x000000076ab08ec8> (a java.lang.ref.ReferenceQueue$Lock)
	at java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:143)
	- locked <0x000000076ab08ec8> (a java.lang.ref.ReferenceQueue$Lock)
	at java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:164)
	at java.lang.ref.Finalizer$FinalizerThread.run(Finalizer.java:209)

"Reference Handler" #2 daemon prio=10 os_prio=31 tid=0x00007fc83b820000 nid=0x3603 in Object.wait() [0x0000700000c37000]
   java.lang.Thread.State: WAITING (on object monitor)
	at java.lang.Object.wait(Native Method)
	- waiting on <0x000000076ab06b68> (a java.lang.ref.Reference$Lock)
	at java.lang.Object.wait(Object.java:502)
	at java.lang.ref.Reference.tryHandlePending(Reference.java:191)
	- locked <0x000000076ab06b68> (a java.lang.ref.Reference$Lock)
	at java.lang.ref.Reference$ReferenceHandler.run(Reference.java:153)

"main" #1 prio=5 os_prio=31 tid=0x00007fc83d001800 nid=0x1703 waiting on condition [0x0000700000219000]
   java.lang.Thread.State: WAITING (parking)
	at sun.misc.Unsafe.park(Native Method)
	- parking to wait for  <0x000000076b2aeb20> (a java.util.concurrent.CountDownLatch$Sync)
	at java.util.concurrent.locks.LockSupport.park(LockSupport.java:175)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer.parkAndCheckInterrupt(AbstractQueuedSynchronizer.java:836)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer.doAcquireSharedInterruptibly(AbstractQueuedSynchronizer.java:997)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer.acquireSharedInterruptibly(AbstractQueuedSynchronizer.java:1304)
	at java.util.concurrent.CountDownLatch.await(CountDownLatch.java:231)
	at org.openjdk.jmh.runner.InfraControlL2.awaitWarmupReady(InfraControl.java:219)
	at org.openjdk.jmh.runner.InfraControl.awaitWarmupReady(InfraControl.java:40)
	at org.openjdk.jmh.runner.BenchmarkHandler.runIteration(BenchmarkHandler.java:349)
	at org.openjdk.jmh.runner.BaseRunner.runBenchmark(BaseRunner.java:263)
	at org.openjdk.jmh.runner.BaseRunner.runBenchmark(BaseRunner.java:235)
	at org.openjdk.jmh.runner.BaseRunner.doSingle(BaseRunner.java:142)
	at org.openjdk.jmh.runner.BaseRunner.runBenchmarksForked(BaseRunner.java:76)
	at org.openjdk.jmh.runner.ForkedRunner.run(ForkedRunner.java:72)
	at org.openjdk.jmh.runner.ForkedMain.main(ForkedMain.java:84)

"VM Thread" os_prio=31 tid=0x00007fc83b819800 nid=0x3403 runnable

"GC task thread#0 (ParallelGC)" os_prio=31 tid=0x00007fc83d00d800 nid=0x2403 runnable

"GC task thread#1 (ParallelGC)" os_prio=31 tid=0x00007fc83c007000 nid=0x2603 runnable

"GC task thread#2 (ParallelGC)" os_prio=31 tid=0x00007fc83c008000 nid=0x2803 runnable

"GC task thread#3 (ParallelGC)" os_prio=31 tid=0x00007fc83c008800 nid=0x2a03 runnable

"GC task thread#4 (ParallelGC)" os_prio=31 tid=0x00007fc83b804000 nid=0x2c03 runnable

"GC task thread#5 (ParallelGC)" os_prio=31 tid=0x00007fc83b80c800 nid=0x2e03 runnable

"GC task thread#6 (ParallelGC)" os_prio=31 tid=0x00007fc83d00e000 nid=0x3003 runnable

"GC task thread#7 (ParallelGC)" os_prio=31 tid=0x00007fc83d00e800 nid=0x3203 runnable

"VM Periodic Task Thread" os_prio=31 tid=0x00007fc83e000000 nid=0x5403 waiting on condition

JNI global references: 361

```
